### ISSUE TITLE
URL creation doesn't ensure strings for parameters

### ISSUE THREAD

**Issue by allwalte** [cloudflare/python-cloudflare#188]
I was trying to write something that does a list on various endpoints, then loops through the returns and deletes each one.  This worked fine for most, but then when I tried it for logpush jobs I got:

```
Traceback (most recent call last):
  File "/Users/awalters/src/it-tools/python/projects/itpy/itpy/cloudflare/bin/remove_added_resources.py", line 87, in <module>
    main()
  File "/Users/awalters/src/it-tools/python/projects/itpy/itpy/cloudflare/bin/remove_added_resources.py", line 76, in main
    endpoint[1].delete(account_id, record_id)
  File "/Users/awalters/Library/Caches/pypoetry/virtualenvs/itpy-vf1dqy_h-py3.10/lib/python3.10/site-packages/CloudFlare/cloudflare.py", line 769, in delete
    return self._do('DELETE', self._parts, [identifier1, identifier2, identifier3, identifier4], params, data)
  File "/Users/awalters/Library/Caches/pypoetry/virtualenvs/itpy-vf1dqy_h-py3.10/lib/python3.10/site-packages/CloudFlare/cloudflare.py", line 233, in do_auth
    return self._call(method, parts, identifiers, params, None, data, files)
  File "/Users/awalters/Library/Caches/pypoetry/virtualenvs/itpy-vf1dqy_h-py3.10/lib/python3.10/site-packages/CloudFlare/cloudflare.py", line 537, in _call
    response_data = self._raw(method, self.headers, parts, identifiers, params, data_str, data_json, files)
  File "/Users/awalters/Library/Caches/pypoetry/virtualenvs/itpy-vf1dqy_h-py3.10/lib/python3.10/site-packages/CloudFlare/cloudflare.py", line 429, in _raw
    [response_type, response_code, response_data] = self._call_network(method,
  File "/Users/awalters/Library/Caches/pypoetry/virtualenvs/itpy-vf1dqy_h-py3.10/lib/python3.10/site-packages/CloudFlare/cloudflare.py", line 271, in _call_network
    url = (self.base_url + '/'
TypeError: can only concatenate str (not "int") to str
```

So in my function where I'm calling the delete, I just cast the id to a string and it's fine.  However, I thought it might be good for the URL formation to ensure strings as well, in places like here:  https://github.com/cloudflare/python-cloudflare/blob/master/CloudFlare/cloudflare.py#L290C21-L293C39

Also it's just really odd that this one resource has the ID as a string, so there might be a chance it would do odd things elsewhere in the code as well.

**Comment by mahtin** [comment]
I think k this issue is that your arguments to ...
```
    endpoint[1].delete(account_id, record_id)
```
... aren't correct. Both should be `str()`. Looks like one is an `int()`.

That said, the error should be way-cleaner. I'll look.

**Comment by mahtin** [comment]
This has been fixed as of `2.19.4`. This issue has existed in all code from the beginning of time; however, the exception thrown wasn't clear. Now all values are converted (if possible) to a string and then processed; you can pass a numeric value (but I really don't recommend it!).

**Comment by mahtin** [comment]
OK - I just went and rechecked this issue and I realize that I just don't handle this specific case very well. For example, should you know the hex value of a `zone_id`, then you should be able to pass this along as an `int()` value. This still fails:
```bash
$ python -c 'import CloudFlare;cf=CloudFlare.CloudFlare();z=cf.zones(0xe87fdd82████████████████████████);print(z["id"],z["name"])'
...
CloudFlare.exceptions.CloudFlareAPIError: Could not route to /client/v4/zones/309044█████████████████████████████████████, perhaps your object identifier is invalid?
$
```
Clearly, there's a decimal vs hex conversion going on now. So passing a hex/int value does not work.

As I said above, I really really don't recommend using `int` values. Stick to `str` values.
